From 4e9f4a4cd6fce12228bac273026300f79c019524 Mon Sep 17 00:00:00 2001
From: nicm <nicm>
Date: Tue, 14 May 2024 09:32:37 +0000
Subject: [PATCH 2/2] Revert part of the change for GitHub issue 3675 because
 it does not work correctly, it was intended to skip lines that are already
 being searched as part of a previous wrapped line but in fact is skipping all
 lines except the last in wrapped lines.

Also revert the search-wrapped-lines option (I didn't realize it was
intended to work around this).
---
 options-table.c |  8 --------
 tmux.1          |  8 --------
 window-copy.c   | 21 ++++-----------------
 3 files changed, 4 insertions(+), 33 deletions(-)

diff --git a/options-table.c b/options-table.c
index 52bea2..f030f2 100644
--- a/options-table.c
+++ b/options-table.c
@@ -383,14 +383,6 @@ const struct options_table_entry options_table[] = {
 	  .text = "Maximum number of commands to keep in history."
 	},
 
-	{ .name = "search-wrapped-lines",
-	  .type = OPTIONS_TABLE_FLAG,
-	  .scope = OPTIONS_TABLE_SERVER,
-	  .default_num = 1,
-	  .text = "Whether to include full wrapped lines when searching for "
-		  "text in copy mode."
-	},
-
 	{ .name = "set-clipboard",
 	  .type = OPTIONS_TABLE_CHOICE,
 	  .scope = OPTIONS_TABLE_SERVER,
diff --git a/tmux.1 b/tmux.1
index cbeb6c..9c0283 100644
--- a/tmux.1
+++ b/tmux.1
@@ -3791,14 +3791,6 @@ each client.
 .It Ic prompt-history-limit Ar number
 Set the number of history items to save in the history file for each type of
 command prompt.
-.It Xo Ic search-wrapped-lines
-.Op Ic on | off
-.Xc
-Defines how
-.Nm
-handles wrapped lines when searching in copy mode.
-When disabled, lines are truncated and searching is faster, but matches may be
-missed.
 .It Xo Ic set-clipboard
 .Op Ic on | external | off
 .Xc
diff --git a/window-copy.c b/window-copy.c
index 6ada5a..830309 100644
--- a/window-copy.c
+++ b/window-copy.c
@@ -3596,13 +3596,10 @@ window_copy_search_jump(struct window_mode_entry *wme, struct grid *gd,
     struct grid *sgd, u_int fx, u_int fy, u_int endline, int cis, int wrap,
     int direction, int regex)
 {
-	u_int			 i, px, sx, ssize = 1;
-	int			 wrapped, found = 0, cflags = REG_EXTENDED;
-	char			*sbuf;
-	regex_t			 reg;
-	struct grid_line	*gl;
-
-	wrapped = options_get_number(global_options, "search-wrapped-lines");
+	u_int	 i, px, sx, ssize = 1;
+	int	 found = 0, cflags = REG_EXTENDED;
+	char	*sbuf;
+	regex_t	 reg;
 
 	if (regex) {
 		sbuf = xmalloc(ssize);
@@ -3619,11 +3616,6 @@ window_copy_search_jump(struct window_mode_entry *wme, struct grid *gd,
 
 	if (direction) {
 		for (i = fy; i <= endline; i++) {
-			gl = grid_get_line(gd, i);
-			if (!wrapped &&
-			    i != endline &&
-			    gl->flags & GRID_LINE_WRAPPED)
-				continue;
 			if (regex) {
 				found = window_copy_search_lr_regex(gd,
 				    &px, &sx, i, fx, gd->sx, &reg);
@@ -3637,11 +3629,6 @@ window_copy_search_jump(struct window_mode_entry *wme, struct grid *gd,
 		}
 	} else {
 		for (i = fy + 1; endline < i; i--) {
-			gl = grid_get_line(gd, i - 1);
-			if (!wrapped &&
-			    i != endline &&
-			    gl->flags & GRID_LINE_WRAPPED)
-				continue;
 			if (regex) {
 				found = window_copy_search_rl_regex(gd,
 				    &px, &sx, i - 1, 0, fx + 1, &reg);
-- 
2.45.2

